pkg/operator/ceph/csi/spec.go

```diff
@@ -392,6 +395,56 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	return nil
 }
 
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	// skip check if running unsupported images is allowed
+	if AllowUnsupported {
+		logger.Infof("Allowing unsupported ceph-csi images -> no version check")
+		return nil
+	}
+
+	ownerRef := metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "Job",
+		Name:       detectCSIVersionName,
+	}
+	timeout := 15 * time.Minute
+
+	logger.Infof("detecting the ceph csi image version for image %s", csiPluginImage)
+	versionReporter, err := cmdreporter.New(
+		clientset, &ownerRef,
+		detectCSIVersionName, detectCSIVersionName, namespace,
+		[]string{"cephcsi"}, []string{"--version"},
+		rookImage, csiPluginImage)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to set up ceph CSI version job. %+v", err)
```

Comment:
`Wrapf` is already wrapping the error. No need for `%+v`.

Reply:
Ah. You are correct!